### PR TITLE
User activity table

### PR DESCRIPTION
**Pull request by NicholasKao1029**
Initial PR for user activity table.

Unsure what columns userActivityTable should have so this PR can help facilitate that discussion

### CODE REVIEW

**Comment by Harin329** [comment]
Off the top of my head I think the columns we should have are like you have done:

user_activity_id
user_id
date

I'm thinking what we could do is add a column like `activity_type`, and this would include things like RECIPE_LIKE, USER_FOLLOW, etc. and would kind of be a table including all activities. With that, we would also need some column like `TARGET_ID`, which would have either the ID of the user followed or the ID of the recipe liked.

Upon typing this out it does seem like it would make sense to have two separate tables for this, however, if we were to ever add more `activity types`, it would be more easily managed under one table, because we can just query by activity type and this would reduce the complexity and number of stored procedures. And I don't think we need to be concerned about too many rows slowing query speed just yet and if we ever do reach that stage we'll probably have a better understanding of what `activities` we need and can split up the table then. 

tldr  something along the lines of `activity_type` and `target_id`

**Comment by NicholasKao1029** [comment]
@Harin329 

Does this mean we would have a corresponding `target_id` for every type inside of `activity_type`

e.g.
USER_FOLLOW -> FK to user_table
RECIPE_LIKE -> FK to recipe_table

**Comment by mebesius** [comment]
I think this is on the right track. @Harin329 brought up a good suggestion for additional columns we might need. Structurally this wouldn't change Harin's suggestion, but for the sake of discussion, another idea for an activity type could be recipe creation. This could also be seen as redundant given the recipes table being tied to the user who authored/created the recipe. 

Probably out of scope (and difficult), but any kind of event we would deem worth tracking for analytics could fit in the "user activity" category as well. Though it would likely need its own table/3rd party tool to realistically add.

**Comment by NicholasKao1029** [comment]
Updated to take in the two different types of activity types and their corresponding FK

**Comment by Harin329** [comment]
I think we'll be okay for recipe creation because we can get that from the recipe table directly, if the recipe row were to have a created_time field. I'm failing to think of any other activity types we might need so I think this is fine.

**Review by NicholasKao1029 — COMMENTED** [review]
UNBLOCK ME

**Review by mebesius — APPROVED** [review]
